package.json

```diff
@@ -1,34 +1,23 @@
 {
   "name": "node-fetch",
-  "version": "3.0.0-beta.9",
+  "version": "3.0.0-beta.10",
   "description": "A light-weight module that brings Fetch API to node.js",
-  "main": "./dist/index.cjs",
-  "module": "./src/index.js",
+  "main": "./src/index.js",
   "sideEffects": false,
   "type": "module",
-  "exports": {
-    ".": {
-      "import": "./src/index.js",
-      "require": "./dist/index.cjs"
-    },
-    "./package.json": "./package.json"
-  },
```

Comment:
I think the correct thing to do is to remove `"main"`, and to add

```js
  "exports": "./src/index.js"
```

Reply:
Don't think we need exports at all?

Reply:
I’m pretty sure we do, how else would the package consumers know what file they should load when importing?

Reply:
From the `main` ofc

fetch-blob@3 is a ESM only package, it dose not include any exports or cjs, only a `main` + `type=module`

Reply:
with type=module, you no longer set `main`, you set `exports`. `main` might still be supported but `exports` is the new way to go, it's going to be expected by lots of tooling out there. When using exports, it will also be easier to define different exports for different environments, such as browser, node, deno, cloudflare workers, etc.

Reply:
From [Sindre's guide](https://github.com/sindresorhus/meta/discussions/15), which is the one I've been following in my repos, one should add `"exports": "./src/index.js"`. I don't have any more info on this though, but what @gr2m said seems right to me 🤔

Reply:
I thought exports was just like some kind of [import map](https://github.com/WICG/import-maps)

but when i read upon it it says:

> In a package’s package.json file, two fields can define entry points for a package: "main" and "exports". The "main" field is supported in all versions of Node.js, but its capabilities are limited: it only defines the main entry point of the package.
> 
> The "exports" field provides an alternative to "main" where the package main entry point can be defined while also encapsulating the package, preventing any other entry points besides those defined in "exports". This encapsulation allows module authors to define a public interface for their package.
> 
> If both "exports" and "main" are defined, the "exports" field takes precedence over "main". "exports" are not specific to ES modules or CommonJS; "main" is overridden by "exports" if it exists. As such "main" cannot be used as a fallback for CommonJS but it can be used as a fallback for legacy versions of Node.js that do not support the "exports" field.

ref: https://nodejs.org/api/packages.html#packages_package_entry_points

So... if you have a exports than you can't simply just use `import('node-fetch/src/response.js')` what if ppl want to do that? it sounds like `main` is sufficient

Reply:
> So... if you have a exports than you can't simply just use `import('node-fetch/src/response.js')`? or i'm wrong?
> what if ppl want to do that? it sounds like `main` is sufficient..?

Importing `node-fetch/src/response.js` is using an internal API that could break any time because it's not tested. If we wanted to make it possible to import the `Response` class selectively we could do

```js
  "exports": {
    ".": "./src/index.js",
    "./response": "./src/response.js"
  },
```

And we should cover the import in our tests.

I haven't seen an ESM package which still uses `"main"`, it feels odd to migrate to `esm` but not use the `exports` key. Did you test that the import of `node-fetch` is working for you? I'll try to give it a try myself